### PR TITLE
Upgrade scancode to  Python 3.7 and 3.8

### LINKED ISSUE #1817: Configuring with Python 3.7 fails due to no matching distribution found for future>=0.16.0
### Description

I'm trying to configure ScanCode (current `develop` branch) with Python 3.7 as part of my work on an [Alpine-based Docker image](https://github.com/sschuberth/docker-files/tree/alpine-miniconda3). However, running `./configure` fails with

```
Collecting future>=0.16.0 (from scancode-toolkit==3.1.2->-r /opt/scancode/etc/conf/requirements_base.txt (line 10))
  ERROR: Could not find a version that satisfies the requirement future>=0.16.0 (from scancode-toolkit==3.1.2->-r /opt/scancode/etc/conf/requirements_base.txt (line 10)) (from versions: none)
ERROR: No matching distribution found for future>=0.16.0 (from scancode-toolkit==3.1.2->-r /opt/scancode/etc/conf/requirements_base.txt (line 10))
* Configuring Python ...
* Installing components ...
Traceback (most recent call last):
  File "/opt/scancode/etc/configure.py", line 552, in <module>
    install_3pp(configs, root_dir, thirdparty_dirs, quiet=quiet)
  File "/opt/scancode/etc/configure.py", line 266, in install_3pp
    run_pip(requirements, root_dir, tpp_dirs, quiet)
  File "/opt/scancode/etc/configure.py", line 292, in run_pip
    call(pcmd, root_dir)
  File "/opt/scancode/etc/configure.py", line 139, in call
    subprocess.check_call(cmd, shell=True, env=dict(os.environ), cwd=root_dir)
  File "/opt/conda/lib/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '"/opt/scancode/bin/pip" install --upgrade --no-index --no-cache-dir --find-links="/opt/scancode/thirdparty" --requirement "/opt/scancode/etc/conf/requirements_base.txt" --requirement "/opt/scancode/etc/conf/dev/requirements_base.txt"' returned non-zero exit status 1.
```

Can this be because the is no `future-*py3*` file in `thirdparty/`?

### System configuration

For bug reports, it really helps us to know:
* What OS are you running on? Ubuntu Linux 18.04
* What version of scancode-toolkit was used to generate the scan file? n/a
* What installation method was used to install/run scancode? From Git.

### PR DESCRIPTION
**Pull request by Abhishek-Dev09**
Upgrade third dependencies for support python 3.7 and python 3.8 .  
After merging this we were able to install scancode with python 3.7 and python 3.8 .
Fix #1817 

Signed-off-by: Abhishek Kumar <abhishek.kasyap09@gmail.com>

### CODE REVIEW

File: thirdparty/restructuredtext_lint-1.3.0-cp37-none-any.whl.ABOUT [hunk 1]
```diff
@@ -0,0 +1,15 @@
+about_resource: restructuredtext_lint-1.3.0-cp37-none-any.ABOUT
+name: restructuredtext_lint
```
Comment (pombredanne) [inline thirdparty/restructuredtext_lint-1.3.0-cp37-none-any.whl.ABOUT]: What is this new library for?
Also the notice URL may be rather https://github.com/twolfson/restructuredtext-lint/blob/1.3.0/UNLICENSE ?

Comment (Abhishek-Dev09) [reply thirdparty/restructuredtext_lint-1.3.0-cp37-none-any.whl.ABOUT]: This library is same as https://github.com/nexB/scancode-toolkit/blob/develop/thirdparty/restructuredtext_lint-1.3.0-cp36-none-any.whl

**Comment by Abhishek-Dev09** [comment]
> Thank you ++
> This obviously needs also the windows and linux wheels.... but before adding more, would you know if there is way to have one wheel that can run on multiple CPython version (3.6, 3.7 and 3.8?)

i have already added but we have to verify that its work or not about each wheel.

**Comment by majurg** [comment]
@Abhishek-Dev09 @pombredanne What is left to do here? I am interested in getting things working on python37+ environments

**Comment by priv-kweihmann** [comment]
> @Abhishek-Dev09 @pombredanne What is left to do here? I am interested in getting things working on python37+ environments

I was about to ask the same - is there any point where a helping hand is needed?

**Comment by pombredanne** [comment]
@majurg @priv-kweihmann
re:
>> What is left to do here? I am interested in getting things working on python37+ environments
> I was about to ask the same - is there any point where a helping hand is needed?

I am on it. But I want first to push a 3.2 tag ASAP

**Comment by ncoghlan** [comment]
@pombredanne With the dependencies that use Cython generated code and put it in the sdist, the only way to build against the stable CPython ABI is to update the source module using a Cython 3.0 pre-release that declares `CYTHON_LIMITED_API`.

**Comment by ncoghlan** [comment]
@pombredanne I saw that you've switched to the pure Python urlpy fork in #1835. Should this PR be closed in favour of that one? Or are there other wheel updates from this PR that should still be included?

**Comment by pombredanne** [comment]
@Abhishek-Dev09 since we are taking a slightly different approach for the 3.7 and 3.8 support, I am closing this PR (though you may want to keep the branch around as some of the ABOUT files an wheel may still be things you want to use in our "pseudo" PyPI index?

**Comment by pombredanne** [comment]
@ncoghlan Re:

>  I saw that you've switched to the pure Python urlpy fork in #1835. Should this PR be closed in favour of that one? Or are there other wheel updates from this PR that should still be included?

Yes, actually we are also building a public PyPI of pre-built wheel for 3.6/.7/.8.... that can be used to install on systems that do not have the native compilation toolchain machinery on hand and installed. 


> With the dependencies that use Cython generated code and put it in the sdist, the only way to build against the stable CPython ABI is to update the source module using a Cython 3.0 pre-release that declares CYTHON_LIMITED_API.

Great to know! We may selectively patch and contribute upstream for some packages with native code then!

**Pull request closed without merging**